### PR TITLE
Online+move

### PR DESCRIPTION
**Pull request by ZyXeM**
Movement through the server

### CODE REVIEW

File: core/src/works/maatwerk/generals/PlayingScreen.java [hunk 1]
```diff
@@ -51,86 +53,90 @@
...
-        map.addCharacter(character20);
+//        Character character18 = new Character(new Race("Test", new Stats()),new Rank(RankName.GENERAL), WeaponClass.CORRUPT, assetManager, new Vector2(17,27));
+//        character18.setWeapon(new Weapon("Axe",2,new Stats(),false,null));
+//        map.addCharacter(character18);
+//        Character character19 = new Character(new Race("Test", new Stats()),new Rank(RankName.GRUNT), WeaponClass.CORRUPT, assetManager, new Vector2(16,28));
+//        character19.setWeapon(new Weapon("Axe",2,new Stats(),false,null));
+//        map.addCharacter(character19);
+//        Character character20 = new Character(new Race("Test", new Stats()), new Rank(RankName.GRUNT), WeaponClass.CORRUPT, assetManager, new Vector2(18,28));
+//        character20.setWeapon(new Weapon("Axe",2,new Stats(),false,null));
+//        map.addCharacter(character20);
```
Comment (REXOTIUM) [inline core/src/works/maatwerk/generals/PlayingScreen.java]: Why is this all outcommented and not removed?

File: core/src/works/maatwerk/generals/PlayingScreen.java [hunk 2]
```diff
@@ -51,86 +53,90 @@
...
 
     private void initializeVolumeControls() {
         multiplexer.addProcessor(new MusicController(game.getMusicManager()));
     }
 
     private void initializeNetworking() {
-        NetworkManager networkManager = new NetworkManager(game);
-        networkManager.connect();
+      //  NetworkManager networkManager = new NetworkManager(game);
+       // networkManager.connect(1);
```
Comment (REXOTIUM) [inline core/src/works/maatwerk/generals/PlayingScreen.java]: Why outcommented and not removed?

File: core/src/works/maatwerk/generals/models/MapManager.java [hunk 3]
```diff
@@ -221,17 +240,25 @@ public void removeCharacter(Character character) {
...
     /**
-     * @param character
+     * @param characterId
      * @param location
      */
-    public void moveCharacter(Character character, Vector2 location) {
+    public void moveCharacter(int characterId, Vector2 location) {
         if (location.x > mapDimensions.x || location.y > mapDimensions.y || location.x < 0 || location.y < 0) {
-            Logger.getLogger(MapManager.class.getName()).log(Level.FINE, "Out of boundaries");
+            com.sun.media.jfxmedia.logging.Logger.logMsg(1, "Out of boundaries");
```
Comment (REXOTIUM) [inline core/src/works/maatwerk/generals/models/MapManager.java]: Why this logger?

File: core/src/works/maatwerk/generals/networking/listeners/UpdateReceiver.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    }
+
+    public void updateDeleteCharacter(Packet packet){
+        DeleteUpdate data = (DeleteUpdate) packet.data;
+        mapmanager.removeCharacter(mapmanager.getCharacterById(data.characterId));
+    }
+
+
+
+
```
Comment (REXOTIUM) [inline core/src/works/maatwerk/generals/networking/listeners/UpdateReceiver.java]: Remove useless enters

File: core/src/works/maatwerk/generals/networking/messages/Message.java [hunk 5]
```diff
@@ -1,4 +1,8 @@
 package works.maatwerk.generals.networking.messages;
 
+@SuppressWarnings("WeakerAccess")
 public abstract class Message {
+
+    public Message() {
```
[the remark this answers is not in the document]
Comment (REXOTIUM) [reply core/src/works/maatwerk/generals/networking/messages/Message.java]: why have this constructor?

File: core/src/works/maatwerk/generals/networking/messages/Packet.java [hunk 6]
```diff
@@ -1,8 +1,10 @@
 package works.maatwerk.generals.networking.messages;
 
 public class Packet {
-    private MessageType type;
-    private Message data;
+
+
```
Comment (REXOTIUM) [inline core/src/works/maatwerk/generals/networking/messages/Packet.java]: Remove useless enters

File: core/src/works/maatwerk/generals/networking/messages/gamestatus/AddCharacterUpdate.java [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+
+    public Character getCharacter() {
+        return character;
+    }
+
+    public void setCharacter(Character character) {
+        this.character = character;
+    }
+
+    Character character;
```
[the remark this answers is not in the document]
Comment (REXOTIUM) [reply core/src/works/maatwerk/generals/networking/messages/gamestatus/AddCharacterUpdate.java]: this

[the remark this answers is not in the document]
Comment (REXOTIUM) [reply core/src/works/maatwerk/generals/networking/messages/gamestatus/AddCharacterUpdate.java]: this

File: core/src/works/maatwerk/generals/networking/messages/gamestatus/AddCharacterUpdate.java [hunk 8]
```diff
@@ -0,0 +1,22 @@
+package works.maatwerk.generals.networking.messages.gamestatus;
+
+import works.maatwerk.generals.models.Character;
+import works.maatwerk.generals.networking.messages.Message;
+
+public class AddCharacterUpdate extends Message {
+
+    public AddCharacterUpdate() {
```
[the remark this answers is not in the document]
Comment (REXOTIUM) [reply core/src/works/maatwerk/generals/networking/messages/gamestatus/AddCharacterUpdate.java]: this

File: core/src/works/maatwerk/generals/networking/messages/gamestatus/AddCharacterUpdate.java [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+        return character;
+    }
+
+    public void setCharacter(Character character) {
+        this.character = character;
+    }
+
+    Character character;
+
+
```
Comment (REXOTIUM) [inline core/src/works/maatwerk/generals/networking/messages/gamestatus/AddCharacterUpdate.java]: Remove useless enters

File: core/src/works/maatwerk/generals/networking/messages/gamestatus/AttackMessage.java [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+
+    public int getAttackY() {
+        return attackY;
+    }
+
+    public void setAttackY(int attackY) {
+        this.attackY = attackY;
+    }
+
+    public int attackY;
```
[the remark this answers is not in the document]
Comment (REXOTIUM) [reply core/src/works/maatwerk/generals/networking/messages/gamestatus/AttackMessage.java]: this

File: core/src/works/maatwerk/generals/networking/messages/gamestatus/AttackMessage.java [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+
+
```
Comment (REXOTIUM) [inline core/src/works/maatwerk/generals/networking/messages/gamestatus/AttackMessage.java]: Remove useless enters

File: core/src/works/maatwerk/generals/networking/messages/gamestatus/DeleteUpdate.java [hunk 12]
```diff
@@ -0,0 +1,19 @@
+package works.maatwerk.generals.networking.messages.gamestatus;
+
+import works.maatwerk.generals.networking.messages.Message;
+
+public class DeleteUpdate extends Message {
+    public DeleteUpdate() {
```
[the remark this answers is not in the document]
Comment (REXOTIUM) [reply core/src/works/maatwerk/generals/networking/messages/gamestatus/DeleteUpdate.java]: this

File: core/src/works/maatwerk/generals/networking/messages/gamestatus/DeleteUpdate.java [hunk 13]
```diff
@@ -0,0 +1,19 @@
...
+
+    public int getCharacterId() {
+        return characterId;
+    }
+
+    public void setCharacterId(int characterId) {
+        this.characterId = characterId;
+    }
+
+    public int characterId;
```
[the remark this answers is not in the document]
Comment (REXOTIUM) [reply core/src/works/maatwerk/generals/networking/messages/gamestatus/DeleteUpdate.java]: this

File: core/src/works/maatwerk/generals/networking/messages/gamestatus/JoinGameMessage.java [hunk 14]
```diff
@@ -7,13 +7,28 @@
...
-    public JoinGameMessage() {
-    }
 
     public JoinGameMessage(String gameId, String sessionId, int userId) {
         this.gameId = gameId;
         this.userId = userId;
         this.sessionId = sessionId;
     }
+
+    public JoinGameMessage () {}
```
Comment (REXOTIUM) [inline core/src/works/maatwerk/generals/networking/messages/gamestatus/JoinGameMessage.java]: No oneliners

File: core/src/works/maatwerk/generals/networking/messages/gamestatus/MoveMessage.java [hunk 15]
```diff
@@ -0,0 +1,44 @@
+package works.maatwerk.generals.networking.messages.gamestatus;
+
+import works.maatwerk.generals.networking.messages.Message;
+
+public class MoveMessage extends Message {
+
+    int x;
```
[the remark this answers is not in the document]
Comment (REXOTIUM) [reply core/src/works/maatwerk/generals/networking/messages/gamestatus/MoveMessage.java]: this

File: core/src/works/maatwerk/generals/networking/messages/gamestatus/MoveMessage.java [hunk 16]
```diff
@@ -0,0 +1,44 @@
...
+    int y;
```
[the remark this answers is not in the document]
Comment (REXOTIUM) [reply core/src/works/maatwerk/generals/networking/messages/gamestatus/MoveMessage.java]: this

File: core/src/works/maatwerk/generals/networking/messages/gamestatus/MoveMessage.java [hunk 17]
```diff
@@ -0,0 +1,44 @@
...
+    int id;
```
[the remark this answers is not in the document]
Comment (REXOTIUM) [reply core/src/works/maatwerk/generals/networking/messages/gamestatus/MoveMessage.java]: this

File: core/src/works/maatwerk/generals/networking/messages/gamestatus/MoveMessage.java [hunk 18]
```diff
@@ -0,0 +1,44 @@
...
+
+
+
+    public MoveMessage(int x, int y, int id) {
+        this.x = x;
+        this.y = y;
+        this.id = id;
+    }
+
+    public MoveMessage() {}
```
Comment (REXOTIUM) [inline core/src/works/maatwerk/generals/networking/messages/gamestatus/MoveMessage.java]: Constructors to the top (lowest amount of arguments first)

Remove useless enters

File: core/src/works/maatwerk/generals/networking/messages/gamestatus/MoveUpdate.java [hunk 19]
```diff
@@ -0,0 +1,41 @@
+package works.maatwerk.generals.networking.messages.gamestatus;
+
+import works.maatwerk.generals.networking.messages.Message;
+
+public class MoveUpdate extends Message {
+    int x;
+    int y;
+    int charachterId;
+
+    public MoveUpdate () {}
```
Comment (REXOTIUM) [inline core/src/works/maatwerk/generals/networking/messages/gamestatus/MoveUpdate.java]: No oneliners

File: core/src/works/maatwerk/generals/networking/messages/gamestatus/MoveUpdate.java [hunk 20]
```diff
@@ -0,0 +1,41 @@
+package works.maatwerk.generals.networking.messages.gamestatus;
+
+import works.maatwerk.generals.networking.messages.Message;
+
+public class MoveUpdate extends Message {
+    int x;
```
[the remark this answers is not in the document]
Comment (REXOTIUM) [reply core/src/works/maatwerk/generals/networking/messages/gamestatus/MoveUpdate.java]: this

File: core/src/works/maatwerk/generals/networking/messages/gamestatus/MoveUpdate.java [hunk 21]
```diff
@@ -0,0 +1,41 @@
...
+    int y;
```
[the remark this answers is not in the document]
Comment (REXOTIUM) [reply core/src/works/maatwerk/generals/networking/messages/gamestatus/MoveUpdate.java]: this

File: core/src/works/maatwerk/generals/networking/messages/gamestatus/MoveUpdate.java [hunk 22]
```diff
@@ -0,0 +1,41 @@
...
+    int charachterId;
```
[the remark this answers is not in the document]
Comment (REXOTIUM) [reply core/src/works/maatwerk/generals/networking/messages/gamestatus/MoveUpdate.java]: this

File: core/src/works/maatwerk/generals/models/MapManager.java (hunk 3 above)
Comment (ZyXeM) [reply core/src/works/maatwerk/generals/models/MapManager.java]: don't know

File: core/src/works/maatwerk/generals/networking/messages/gamestatus/AttackUpdate.java [hunk 23]
```diff
@@ -0,0 +1,30 @@
...
+    public int getTargetId() {
+        return targetId;
+    }
+
+    public void setTargetId(int targetId) {
+        this.targetId = targetId;
+    }
+
+
+
```
Comment (REXOTIUM) [inline core/src/works/maatwerk/generals/networking/messages/gamestatus/AttackUpdate.java]: AAAH enters!!

File: core/src/works/maatwerk/generals/networking/messages/gamestatus/AttackUpdate.java [hunk 24]
```diff
@@ -0,0 +1,30 @@
...
+    public int getCharacterId() {
+        return characterId;
+    }
+
+    public void setCharacterId(int characterId) {
+        characterId = characterId;
+    }
+
+
+
```
Comment (REXOTIUM) [inline core/src/works/maatwerk/generals/networking/messages/gamestatus/AttackUpdate.java]: MORE ENTERS!!

File: core/src/works/maatwerk/generals/networking/messages/gamestatus/MoveUpdate.java [hunk 25]
```diff
@@ -3,9 +3,9 @@
...
 
 public class MoveUpdate extends Message {
-    int x;
-    int y;
-    int charachterId;
+   private int x;
+   private int y;
+   private int charachterId;
 
     public MoveUpdate () {}
```
Comment (REXOTIUM) [inline core/src/works/maatwerk/generals/networking/messages/gamestatus/MoveUpdate.java]: Still a oneliner

**Pull request closed without merging**